[Nikki Than Win] iP

thanwinnikki opened the pull request:
# Duke Assistant
> "Our deepest fear is not that we are inadequate. Our deepest feat is that we are powerful beyond measure."
> -Marianne Williamson [(analysis)](https://poemanalysis.com/marianne-williamson/our-deepest-fear/)

Duke Assistant is now here to help you unlock your full potential. Duke Assistant can help you,

* Keep track of your tasks
* Check them as complete/incomplete
* Find tasks based on key words

You can try the following key words,

1. todo (task_name)
2. deadline (task_name) /by (dd/mm/yyyy) (HHmm) 
3. event (task_name) /at (time) 
4. list
5. delete (task_number)

It is really that ~~simple~~ **SIMPLE**!

Version Releases:

- [x] v 0.1
- [ ] v Nearly Done
- [ ] COMPLETED

Currently this is how Duke Assistant is started in java.

```javascript
public static void main(String[] args) throws InputError {
    new Duke().run();
}
```

We use the `<UI>` class to handle interactions with the users, go on check it out 😉

Commit:
Add Gradle support

thanwinnikki committed:
Added "Greet, Echo, Exit" features to DukeBot.

thanwinnikki committed:
Added "Add, List" functions for DukeBot.

thanwinnikki committed:
Added Task class and marking as done feature implemented.

thanwinnikki committed:
Added ToDo, Event and Deadline classes with working implementation.

thanwinnikki committed:
Added automated text UI testing.

thanwinnikki committed:
Added user input error handling to bot.

thanwinnikki committed:
Added delete function to chat bot.

thanwinnikki committed:
Improved error catching for user input.

thanwinnikki committed:
Add code improvements and readibility.

thanwinnikki committed:
Add correct indentations to ChatBot file.

thanwinnikki committed:
Add file read and file save to Duke Bot.

thanwinnikki committed:
Add datetime format to deadline task.

thanwinnikki committed:
Clean code for Level-8.

thanwinnikki committed:
Merge branch 'branch-Level-7'

thanwinnikki committed:
Merge branch 'branch-Level-8'

thanwinnikki committed:
Organise code for better OOP.

thanwinnikki committed:
Add files to duke package.

thanwinnikki committed:
Add JUnit tests.

thanwinnikki committed:
Create jar file artifact.

thanwinnikki committed:
Add JavaDoc comments to at least half of the classes and methods.

thanwinnikki committed:
Change code to comply with coding standard.

thanwinnikki committed:
Add Find function to duke bot.

thanwinnikki committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/DeadLine.java
#	src/main/java/duke/Parser.java

thanwinnikki committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Task.java

thanwinnikki committed:
Clean up code.

thanwinnikki committed:
Merge remote-tracking branch 'origin/add-gradle-support'

thanwinnikki committed:
Add gradle support to duke.

thanwinnikki committed:
Add checkstyle.

thanwinnikki committed:
Add check style.

thanwinnikki committed:
Update code according to checkstyle.

thanwinnikki committed:
Add additional changes according to Check Style.

thanwinnikki committed:
Merge branch 'A-CheckStyle'

thanwinnikki committed:
Clean up code to add Duke Commands next.

thanwinnikki committed:
Implement working GUI for duke bot.

thanwinnikki committed:
Merge branch 'Level-10'

chaiwanlin commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
```
Would it make the import statements cleaner if different import packages are separated by newlines?

chaiwanlin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,308 @@
+        }
+        return response;
+    }
+    @Override
```
Adding a new line before this method may make it look neater!

chaiwanlin reviewed (commented):
LGTM. Well documented and neat code :+1:!

thanwinnikki committed:
Fix code according to peer code review.

sheehui commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,45 @@
+ * @author Nikki
+ */
+public class Task implements Serializable {
+    protected boolean complete;
```
Boolean variable should be named to sound like boolean values 

```java
protected boolean isComplete;
```

sheehui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+    /**
+     * Creates a task list object.
+     *
+     * * @param list Lists of tasks as an Array List.
```
I think there was a minor error here with the extra *.

sheehui commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+
+    String addTodo(String str) {
+        String response = "";
+        System.out.println("Alrighty! I have added this task:");
+        taskList.add(new ToDo(str.substring(5)));
+        response = "Alrighty! I have added this task:\n" + "   " + taskList.get(taskList.size() - 1).printTask() + "\n"
+                + "Now you have " + taskList.size() + " task(s) in total!";
+        return response;
+    }
+
+    String addDeadline(String str) {
+        String response = "";
+        taskList.add(new DeadLine(str.substring(9, str.indexOf("/") - 1), str.substring(str.indexOf("/") + 4)));
+        response = "Alrighty! I have added this task:\n" + "   " + taskList.get(taskList.size() - 1).printTask()
+                + "\n" + "Now you have " + taskList.size() + " task(s) in total!";
+        return response;
+    }
+
+    String addEvent(String str) {
+        String response = "";
+        taskList.add(new Event(str.substring(6, str.indexOf("/") - 1), str.substring(str.indexOf("/") + 4)));
+        response = "Alrighty! I have added this task:\n" + "   " + taskList.get(taskList.size() - 1).printTask()
+                + "\n" + "Now you have " + taskList.size() + " task(s) in total!";
+        return response;
+    }
+
+    String deleteItem(int index) throws InputError {
+        String response = "";
+        try {
+            if (index > taskList.size()) {
+                throw new InputError("Invalid Number");
+            }
+            Task removed = taskList.remove(index - 1);
+            response = "Alrighty! I have deleted this task:\n" + "   " + removed.printTask() + "\n"
+                    + "Now you have " + taskList.size() + " task(s) in total!";
+        } catch (InputError e) {
+            response = ui.errorMessage(e);
+        }
+        return response;
+    }
```
We should always use access modifiers for our methods!

sheehui commented:
Good job overall! I really like how well documented your code is. Perhaps you may consider adding more packages within duke package for easier understanding/logic (i.e. DialogBox and MainWindow can go under a ui package etc.).

nicolej2122 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
```
Could add comments for the public method

nicolej2122 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+     * @throws InputError If user inputs are invalid or unrecognised.
+     */
+    public void run() throws InputError {
+        boolean byeMessage = false;
```
Can change the boolean variable name to adhere to the coding standard

nicolej2122 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,314 @@
+     * @param str User input.
+     * @return True if bye command. False if not bye command.
+     */
+    private boolean byeChecker(String str) {
```
Can change the method name to isBye instead

nicolej2122 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,314 @@
+     * @param str User input.
+     * @return True if list command. False if not list command.
+     */
+    private boolean listChecker(String str) {
```
Can change the boolean method name to isList

nicolej2122 reviewed (commented):
Looks good! Just some minor changes with regards to the coding standard :)

thanwinnikki committed:
Add assert statements to code

There are many assumptions made in the execution of this code. In
some cases, if these assumptions fails, it will cause the program
to fail.

Making sure these assumptions are checked improves the code quality.

Let's make sure that these assumptions are checked to avoid any
mistakes by the programmer.

thanwinnikki committed:
Improve code quality

There are many lines of code that are not readable to other
readers.

Cleaning up the code in accordance to the textbook vastly improves
readability.

As a step towards better code quality, add the appropriate javadocs,
abstract methods to the same level and eliminate use of magic
numbers.

thanwinnikki committed:
Merge pull request #1 from thanwinnikki/A-Assertions

Add assert statements to code

thanwinnikki committed:
Merge branch 'master' into A-CodeQuality

thanwinnikki committed:
Merge pull request #2 from thanwinnikki/A-CodeQuality

Improve code quality

thanwinnikki committed:
Improve code readability.

thanwinnikki committed:
Add help command to display available commands.

thanwinnikki committed:
Merge branch 'C-Help'

thanwinnikki committed:
Remove unecessary import line.

thanwinnikki committed:
Remove unnecessary code.

thanwinnikki committed:
Remove unnecessary code and clean up code.

thanwinnikki committed:
Add product screen shot.

thanwinnikki committed:
Set theme jekyll-theme-minimal

thanwinnikki committed:
Add user guide to duke bot.

thanwinnikki committed:
Merge branch 'master' of https://github.com/thanwinnikki/ip

thanwinnikki committed:
Update main class name to correct name.

thanwinnikki committed:
Rename boolean in Task class according to naming convention.

thanwinnikki committed:
Remove unnecessary code.

thanwinnikki committed:
Chang name to correct naming.
